week-3/Homework/mandatory/7-password-validator.js

```diff
@@ -22,8 +22,20 @@ PasswordValidationResult=  [false, false, false, false, true]
 
 */
 
-function validatePasswords(passwords) {
+// function validPassword(password, index, arr) {
+//   let pwd = arr.map(password => password.includes(password.match(/^(?=.*[\d])(?=.*[A-Z])(?=.*[a-z])(?=.*[!@#$%^&*])[\w!@#$%^&*]{5,}$/)));
+//   let prePwd = arr.indexOf(password) === index;
+//   return ( pwd && prePwd);
+// }
+function validPassword(password, index, arr) {
+  let pwd = /^(?=.*[\d])(?=.*[A-Z])(?=.*[a-z])(?=.*[!@#$%^&*])[\w!@#$%^&*]{5,}$/;
```

Comment:
This is a pretty complicated looking regular expression - can you talk through what it does? What does each part of it do?

It can often be easier to use a few small regular expressions, rather than one big one - that can make it easier to see what's going on to people reading the code. Could you think of a few different regular expressions, the `test` of which you could `&&` together to make sure they're all matched by the string?